devito/dle/parallelizer.py

```diff
@@ -90,13 +106,17 @@ class Ompizer(object):
     compilation time (e.g., this may happen when DefaultDimensions are used).
     """
 
+    CHUNKSIZE_NONAFFINE = 3
+    """
+    Coefficient to adjust the chunk size in parallelized non-affine Iterations.
+    """
+
     lang = {
-        'for-static': lambda i: c.Pragma('omp for collapse(%d) schedule(static)' % i),
-        'for-static-1': lambda i: c.Pragma('omp for collapse(%d) schedule(static,1)' % i),
-        'for-dynamic-1': lambda i: c.Pragma('omp for collapse(%d) schedule(dynamic,1)'
-                                            % i),
-        'par-for': lambda i, j: c.Pragma('omp parallel for collapse(%d) '
-                                         'schedule(static,1) num_threads(%d)' % (i, j)),
+        'for': lambda i, cs:
+            c.Pragma('omp for collapse(%d) schedule(dynamic,%s)' % (i, cs)),
+        'par-for': lambda i, cs, nt:
+            c.Pragma('omp parallel for collapse(%d) schedule(dynamic,%s) num_threads(%d)'
```

Comment:
is `dynamic` really always better?

Reply:
I couldn't find a single use case in which it wasn't .

In fact, I was fairly shocked: both tti and isotropic acoustic on my workstation ran about 10% faster with dynamic

on the KNL is the same story

Reply:
Wow really, for any threads/space_order/cores...? That's surprising.

Reply:
well I tried a few, and different grid shapes too...

space orders -- I tried 4 and 16 I think

So not a complete study admittedly, but consistent speedups yeah. Also, in the PR description I asked
> Would appreciate some performance testing

😎